crates/check/src/solution.rs

```diff
@@ -356,6 +411,92 @@ pub fn check_set_state_mutations(set: &SolutionSet) -> Result<(), InvalidSolutio
     Ok(())
 }
 
+/// Check the given solution set against the given predicates and
+/// and compute the post state mutations for this set.
+#[cfg_attr(feature = "tracing", tracing::instrument(skip_all))]
+pub async fn check_and_compute_solution_set<SA, SB>(
+    pre_state: &SA,
+    post_state: &SB,
+    solution_set: SolutionSet,
```

Comment:
Ahhhh interesting that we can even still provide some other `post_state` and an existing WIP solution set when computing more solutions!

I guess in the case that we want to compute some new solutions from scratch, we could provide an empty `solution_set` and either some empty state to the `post_state`, or pass the pre-state to both the `pre_state` and `post_state` arguments.

Reply:
We want to remove post state but that's in a separate PR